File: sdk/src/test/java/com/mparticle/MockResources.java
package com.mparticle;

/**
 * Created by sdozor on 3/23/15.
 */
public class MockResources extends android.test.mock.MockResources {
    public static String TEST_APP_KEY = "the app key";
    public static String TEST_APP_SECRET = "the app secret";

    @Override
    public int getIdentifier(String name, String defType, String defPackage) {
        if (name.equals("mp_key")){
            return 1;
        }else if (name.equals("mp_secret")){
            return 2;
        }

        return 0;
    }

    @Override
    public String getString(int id) throws NotFoundException {
        switch (id){
            case 1:
                return TEST_APP_KEY;
            case 2:
                return TEST_APP_SECRET;

        }
        return null;
    }

    @Override
    public String getString(int id, Object... formatArgs) throws NotFoundException {
        return super.getString(id, formatArgs);
    }
}

File: kits/braze/braze-43/src/test/kotlin/com/mparticle/kits/mocks/MockResources.kt
package com.mparticle.kits.mocks

import android.content.res.Resources
import android.content.res.Resources.NotFoundException
import kotlin.Throws

class MockResources : Resources(null, null, null) {
    override fun getIdentifier(
        name: String,
        defType: String,
        defPackage: String,
    ): Int {
        if (name == "mp_key") {
            return 1
        } else if (name == "mp_secret") {
            return 2
        }
        return 0
    }

    @Throws(NotFoundException::class)
    override fun getString(id: Int): String {
        when (id) {
            1 -> return testAppKey
            2 -> return testAppSecret
        }
        return ""
    }

    @Throws(NotFoundException::class)
    override fun getString(
        id: Int,
        vararg formatArgs: Any,
    ): String = super.getString(id, *formatArgs)

    companion object {
        var testAppKey = "the app key"
        var testAppSecret = "the app secret"
    }
}
